core/tx_pool.go

```diff
@@ -1419,6 +1419,8 @@ func (pool *TxPool) demoteUnexecutables() {
 				pool.enqueueTx(hash, tx)
 			}
 			pendingGauge.Dec(int64(len(gapped)))
+			// This might happen in a reorg, so log it to the metering
+			blockReorgInvalidatedTx.Mark(int64(len(gapped)))
```

Comment:
Curious what's the meaning of this meter?

Reorg happens, the "previously confirmed" transaction with nonce N is reverted, then it will cascade invalidate all following transactions.
But usually the double spend attack is: the original transaction with nonce N is confirmed, but after a thousand blocks later the nonce N is replaced with another transaction. So I'm not sure whether it's a useful meter.

Reply:
And maybe we can use a meter defined in the txpool to log these data

Reply:
I think what this logs is tx that were part of the chain before a reorg and after a reorg became invalidated. Logging them can give us an indication if someone sends transactions that are in the reorg but not in the chain I think.
See:
https://gist.github.com/AFDudley/410c0cd1c302b62b13644e06df3bf4eb